src/ccl_core.c

```diff
@@ -306,26 +306,24 @@ TASK: fill parameters not set by ccl_parameters_create with some initial values
 Omega_g = (Omega_g*h^2)/h^2 is the radiation parameter; "g" is for photons, as in CLASS
 T_CMB: CMB temperature in Kelvin
 Omega_l: Lambda
-A_s: amplitude of the primordial PS, enforced here to initially set to NaN
-sigma8: variance in 8 Mpc/h spheres for normalization of matter PS, enforced here to initially set to NaN
+A_s: amplitude of the primordial PS
+sigma8: variance in 8 Mpc/h spheres for normalization of matter PS
 z_star: recombination redshift
  */
 void ccl_parameters_fill_initial(ccl_parameters * params, int *status)
 {
   // Fixed radiation parameters
   // Omega_g * h**2 is known from T_CMB
-  params->T_CMB =  ccl_constants.T_CMB;
   // kg / m^3
   double rho_g = 4. * ccl_constants.STBOLTZ / pow(ccl_constants.CLIGHT, 3) * pow(params->T_CMB, 4);
   // kg / m^3
   double rho_crit =
     ccl_constants.RHO_CRITICAL *
     ccl_constants.SOLAR_MASS/pow(ccl_constants.MPC_TO_METER, 3) *
     pow(params->h, 2);
-  params->Omega_g = rho_g/rho_crit;
 
   // Get the N_nu_rel from Neff and N_nu_mass
-  params->N_nu_rel = params->Neff - params->N_nu_mass * pow(ccl_constants.TNCDM, 4) / pow(4./11.,4./3.);
+  params->N_nu_rel = params->Neff - params->N_nu_mass * pow(params->T_ncdm, 4) / pow(4./11.,4./3.);
```

Comment:
Isn't all of this happening in python now?

Reply:
Yes, but halofit and musigma at the C-level use these to create copy of the cosmologies.